src/StoredList.js

```diff
@@ -0,0 +1,37 @@
+class StoredList {
```

Comment:
Since we are already using [`localForage`](https://github.com/localForage/localForage) in `aragon.js` I propose to expose the same interface. Maybe we don't need the extra abstraction for locally stored arrays and we should use the `localForage` tool directly.

Benefits: simplified _promise-able_ api (`setItem` and `getItem`)

Reply:
After reviewing how the `StoredList` class is being used may I suggest you use a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) it'll help with keeping unique items and it'll also make it easier to add/remove elements.

Reply:
> I would rather avoid adding another dependency

Can I suggest to use the same interface at least? I think we can benefit from the simplicity of the two methods (and the option to use them as promises).

Reply:
I’m not sure I understand why we need this, it’s not exposed anyway? Or do you see it as something that would be exposed? And using it with promises shouldn’t be a problem since it’s sync to start with.

I tried moving to `Set` but it’s probably not really worth it:

- We have to convert it to an array every time a change is made to it (to save it as a JSON string).
- There is no equivalent to `Array.prototype.slice` in `Set`, so removing by index would require an extra `Set → array → Set` conversion.
- We might want to store duplicate values in a `StoredList`.

What do you think?

Reply:
> We might want to store duplicate values in a StoredList. *1*

I had not realized this was required. In which case, yeah, `Set` won't help.

> We have to convert it to an array every time a change is made to it (to save it as a JSON string).

True. I agree we don't need `localForage` here, but it serializes data for us so that we can send and retrieve more than `String`s.

> There is no equivalent to Array.prototype.slice in Set, so removing by index would require an extra Set → array → Set conversion.

I thought you might want to use a structure like:

```js
var list = new Set([
  ['unique_id', { /* data */ }],
  ...
])
```
Which allows you to `list.delete('some_id')` but I see you want to use the index and from *1* that you want to store duplicates.

Thank you for addressing all my ideas/concerns here @bpierre